src/libraries/Directory.Build.props

```diff
@@ -49,6 +49,7 @@
  
     <!-- Initialize BuildSettings from the individual properties if it wasn't already explicitly set -->
     <Configuration Condition="'$(Configuration)'==''">Debug</Configuration>
+    <BuildTargetFramework Condition="'$(BuildAllProjects)' != 'true' and '$(TargetFramework)' != ''">$(TargetFramework)</BuildTargetFramework>
```

Comment:
Should we set this in the sdk instead?

Reply:
```BuildAllProjects``` flag is specific to runtime repo`s vertical build.
I dont see a use case of adding this to sdk as the sdk will never be use this property when TargetFramework is set.

Reply:
Can you please clarify the purpose of `BuildTargetFramework`?

Reply:
```BuildTargetFramework``` is a parameter for the outerbuild to choose the best target framework for the innerbuild.